### PR TITLE
[WIP] Feature/search bar results

### PR DESCRIPTION
**Pull request by TanguyDuponchelle**
Hello,
SearchBar Autosuggestion has been made. Still have issue on Submit (JSX answer).

### CODE REVIEW

**Review by FPaurisse — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -3,6 +3,8 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "autocomplete": "0.0.1",
```
Comment (Aebrathia) [inline package.json]: Remove dead dependency. Also, this dependency is not a good choice : it is very little used and has not received any updates in 8 years.

File: src/App.js [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
-      </section>
-    </div>
-  );
-}
+class App extends React.Component{
+  constructor(props){
+    super(props)
+    this.state = {
+      aoc : ''    
+    }
```
Comment (Aebrathia) [inline src/App.js]: Dead code. You don't need a state. Here App is only presentational, always prefer functional components when possible.

File: src/components/Resultat.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+class Resultat extends React.Component{
+    render(){
```
Comment (Aebrathia) [inline src/components/Resultat.js]: Prefer functions when you don't need a state. It is simpler to write, to read and has better performances.

File: src/components/SearchBar.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
+import Autosuggest from 'react-autosuggest';
+import React from 'react';
+import './css/SearchBar.css'
```
Comment (Aebrathia) [inline src/components/SearchBar.js]: You should put SearchBar.css and SearchBar.js together in a SearchBar folder, like this :
- `/src/components/SearchBar/SearchBar.js`
- `/src/components/SearchBar/SearchBar.css`

File: src/components/SearchBar.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  constructor(props) {
+      super(props);
+      this.state = {
+      value: '',
+      suggestions: [],
+      aoc : [],
+      hasSubmitted : false
+      };
+  this.getResultat = this.getResultat.bind(this)
+  }
```
Comment (Aebrathia) [inline src/components/SearchBar.js]: Fix indentation

File: src/components/SearchBar.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import axios from 'axios';
+import Resultat from './Resultat';
+
+const renderSuggestion = suggestion => (
+  <div>
+    {suggestion}
+  </div>
+);
+
+class SearchBar extends React.Component {
```
Comment (Aebrathia) [inline src/components/SearchBar.js]: Use `extends Component`

File: src/components/SearchBar.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+          />
+          <button type="submit">submit</button>
+        </form>
+        {this.state.hasSubmitted && <Resultat />}
+      </div>
+    );
+  }
+}
+
+export default SearchBar
```
Comment (Aebrathia) [inline src/components/SearchBar.js]: Add missing newline

File: src/components/css/SearchBar.css [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+.react-autosuggest__input{
+    width: 100%;
+}
+
+
+
+@media screen and (max-width: 768px) {
+    
+}
```
Comment (Aebrathia) [inline src/components/css/SearchBar.css]: Remove dead code and add missing newline

File: src/components/css/SearchBar.css [hunk 9]
```diff
@@ -0,0 +1,15 @@
+.SearchBar--suggestion{
+    width: 100%;
+    background-color:rgba(231, 231, 231, 0.671);
+
+}
+
+.react-autosuggest__input{
+    width: 100%;
+}
```
Comment (Aebrathia) [inline src/components/css/SearchBar.css]: Fix indentation and spacings.

Weird naming convention, what do you use ? BEM ?

File: package.json (hunk 1 above)
Comment (Aebrathia) [reply package.json]: Also, `react-autosuggest` is missing from dependencies. You probably used `npm install react-autosuggest --no-save`.

File: src/components/Resultat.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+class Resultat extends React.Component{
```
Comment (Aebrathia) [inline src/components/Resultat.js]: Code in english, not in French

Comment (Aebrathia) [reply src/components/Resultat.js]: Use `extends Component`.

File: src/components/Resultat.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+    render(){
+        return( 
+            <div>
+                halo
+            </div>
+        )
+    }
+}
+export default Resultat
```
Comment (Aebrathia) [inline src/components/Resultat.js]: Rename in english and add missing newline

File: src/components/SearchBar.js [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+          <Autosuggest
+            name = "AOC_Searched"
+            id = "AOC_Searched"z
+            suggestions={suggestions}
+            onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
+            onSuggestionsClearRequested={this.onSuggestionsClearRequested}
+            getSuggestionValue={this.getSuggestionValue}
+            renderSuggestion={renderSuggestion}
+            inputProps={inputProps}
+          />
```
Comment (Aebrathia) [inline src/components/SearchBar.js]: All your difficulties and bugs come from this package which is a bit awkward.

I suggest that you drop it and write the autocompletion yourself and/or that you spend time with me in order to explain you this component.

**Review by Aebrathia — CHANGES_REQUESTED** [review]

**Pull request closed without merging**